ee/management/commands/migrate_clickhouse.py

```diff
@@ -21,21 +27,36 @@ def add_arguments(self, parser):
         parser.add_argument(
             "--plan", action="store_true", help="Shows a list of the migration actions that will be performed."
         )
+        parser.add_argument(
+            "--print-sql",
+            action="store_true",
+            help="Only use with --plan. Also prints SQL for each migration to be applied.",
+        )
 
     def handle(self, *args, **options):
+        for index, host in enumerate(CLICKHOUSE_MIGRATION_HOSTS_HTTP_URLS):
+            print(f"Updating host {host} ({index + 1}/{len(CLICKHOUSE_MIGRATION_HOSTS_HTTP_URLS)})")
+            self.migrate(host, options)
+
+    def migrate(self, host, options):
         database = Database(
             CLICKHOUSE_DATABASE,
-            db_url=CLICKHOUSE_HTTP_URL,
+            db_url=host,
             username=CLICKHOUSE_USER,
             password=CLICKHOUSE_PASSWORD,
             verify_ssl_cert=False,
         )
 
         if options["plan"]:
             print("List of clickhouse migrations to be applied:")
-            for migration_name in self.get_migrations(database, options["upto"]):
+            migrations = list(self.get_migrations(database, options["upto"]))
+            for migration_name, operations in migrations:
                 print(f"Migration would get applied: {migration_name}")
-            else:
+                for op in operations:
+                    sql = getattr(op, "_sql")
+                    if options["print_sql"] and sql is not None:
+                        print(indent("\n\n".join(sql), "    "))
+            if len(migrations) == 0:
```

Comment:
why did you change the `else` to `if len(migrations) == 0`?

Reply:
Previous version was slightly buggy - `for` loop `else` fires always after end (unless you break) which was not what I intended. https://book.pythontips.com/en/latest/for_-_else.html